rustler/src/types/truthy.rs

```diff
@@ -0,0 +1,24 @@
+//!
+//! A type to represent a truthy value.
+//!
+//! In Elixir, a term which does not equal `:false` or `:nil` is considered to be
+//! truthy. This does not cleanly map to Rust's `bool` type. To distinguish between
+//! `bool` and a truthy value, the newtype `Truthy` can be used.
+//!
+
+use crate::types::atom;
+use crate::{Decoder, Encoder, Env, NifResult, Term};
+
+pub struct Truthy(bool);
```

Comment:
I really dislike adding this additional layer of abstraction. The point is not to model Elixir directly, but to be able to map directly to Rust types.

I would expect that if I tried to decode an Elixir value as `bool` that is not `:true` or `:false`, it would fail.

This is an awkward abstraction that does not provide the intent I'm looking for

Reply:
Maybe I'm misunderstanding this. Is this just to provide the option to decode `Truthy` values, but not mandate them?

Looking at the test code, it looks like you're adding a feature and fixing the bug in #214